src/components/MultiCombobox/MultiCombobox.tsx

```diff
@@ -356,17 +363,28 @@ function MultiCombobox<Item>({
               isOpen={isOpen && results.length > 0}
               {...getMenuProps()}
             >
-              {results.map(item => (
-                <MenuItem
-                  {...getItemProps({ item, disabled: disableItem(item) })}
-                  as="li"
-                  listStyle="none"
-                  key={itemToString(item)}
-                  selected={item === selectedItem}
-                >
-                  {itemToString(item)}
-                </MenuItem>
-              ))}
+              {itemToGroup ? (
```

Comment:
I don't like the presence of this conditional. I feel we should always render `MenuItemGroups`, but just rename the component to `ComboboxItems` or something.

In there you'll render `MenuItem` elements either wrapped in groups or not